### PR TITLE
cleanup clippy lints

### PR DESCRIPTION
**Pull request by montanalow**

### CODE REVIEW

File: pgml-extension/pgml_rust/src/api.rs [hunk 1]
```diff
@@ -133,34 +136,39 @@ fn deploy(
...
+            predicate,
+            "\nAND algorithm::TEXT = '{}'",
+            algorithm.to_string().as_str()
+        );
     }
     match strategy {
         Strategy::best_score => match task {
             Task::regression => {
-                sql += &format!("{predicate}\nORDER BY models.metrics->>'r2' DESC NULLS LAST");
+                let _ = write!(
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/api.rs:154]: What come on clippy, this seems totally fine.

File: pgml-extension/pgml_rust/src/api.rs [hunk 2]
```diff
@@ -230,10 +238,7 @@ fn load_dataset(
     limit: Option<default!(i64, "NULL")>,
 ) -> impl std::iter::Iterator<Item = (name!(table_name, String), name!(rows, i64))> {
     // cast limit since pgx doesn't support usize
-    let limit: Option<usize> = match limit {
-        Some(limit) => Some(limit.try_into().unwrap()),
-        None => None,
-    };
+    let limit: Option<usize> = limit.map(|limit| limit.try_into().unwrap());
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/api.rs:241]: Weird.

File: pgml-extension/pgml_rust/src/lib.rs [hunk 3]
```diff
@@ -66,7 +66,7 @@ fn model_predict_batch(model_id: i64, features: Vec<f32>, num_rows: i32) -> Vec<
 
     match guard.get(&model_id) {
         Some(data) => {
-            let bst = Booster::load_buffer(&data).unwrap();
+            let bst = Booster::load_buffer(data).unwrap();
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/lib.rs:69]: This was a clippy warning? Should of been a compilation error...

File: pgml-extension/pgml_rust/src/orm/dataset.rs [hunk 4]
```diff
@@ -86,7 +90,7 @@ pub fn load_diabetes(limit: Option<usize>) -> (String, i64) {
         None => diabetes.num_samples,
     };
     for i in 0..limit {
-        let age = diabetes.data[(i * diabetes.num_features) + 0];
+        let age = diabetes.data[(i * diabetes.num_features)];
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/dataset.rs:93]: Hah!

File: pgml-extension/pgml_rust/src/orm/dataset.rs [hunk 5]
```diff
@@ -153,6 +157,7 @@ pub fn load_digits(limit: Option<usize>) -> (String, i64) {
         let target = digits.target[i];
         // shape the image in a 2d array
         let mut image = vec![vec![0.; 8]; 8];
+        #[allow(clippy::needless_range_loop)] // x & y are in fact used
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/dataset.rs:160]: Oh no, don't get me started with these haha! Rubocop was plenty, thank you

File: pgml-extension/pgml_rust/src/orm/estimator.rs [hunk 6]
```diff
@@ -13,15 +13,16 @@ use crate::orm::Algorithm;
 use crate::orm::Dataset;
 use crate::orm::Task;
 
+#[allow(clippy::type_complexity)]
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/estimator.rs:16]: Clippy needs to get real.

File: pgml-extension/pgml_rust/src/orm/estimator.rs [hunk 7]
```diff
@@ -40,33 +41,26 @@ pub fn find_deployed_estimator_by_project_name(name: &str) -> Arc<Box<dyn Estima
...
-    let task = Task::from_str(
-        &task.expect(
-            format!(
-                "Project {} does not have a trained and deployed model.",
-                name
-            )
-            .as_str(),
-        ),
-    )
+    let task = Task::from_str(&task.unwrap_or_else(|| {
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/estimator.rs:44]: Ok what's up with these. Why does not like `expect`?

File: pgml-extension/pgml_rust/src/orm/snapshot.rs [hunk 8]
```diff
@@ -113,7 +113,7 @@ impl Snapshot {
         Spi::connect(|client| {
             let parts = self
                 .relation_name
-                .split(".")
+                .split('.')
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/snapshot.rs:116]: Ummm. Those are completely different data types if you ask me.

File: pgml-extension/pgml_rust/src/orm/snapshot.rs [hunk 9]
```diff
@@ -279,7 +279,7 @@ impl Snapshot {
                 (num_rows as f32 * self.test_size).round() as usize
             };
             let num_train_rows = num_rows - num_test_rows;
-            if num_train_rows <= 0 {
+            if num_train_rows == 0 {
```
Comment (levkk) [inline pgml-extension/pgml_rust/src/orm/snapshot.rs:282]: Clippy never heard of off-by-one errors? == 0 always made me paranoid.

File: pgml-extension/pgml_rust/src/api.rs (hunk 1 above)
Comment (montanalow) [reply pgml-extension/pgml_rust/src/api.rs:154]: format! does an extra allocation compared to write... 🤷‍♂️

File: pgml-extension/pgml_rust/src/api.rs (hunk 2 above)
Comment (montanalow) [reply pgml-extension/pgml_rust/src/api.rs:241]: yeah, turns out you can map things like this...

File: pgml-extension/pgml_rust/src/lib.rs (hunk 3 above)
Comment (montanalow) [reply pgml-extension/pgml_rust/src/lib.rs:69]: The compiler handles the extra dereference for you.

File: pgml-extension/pgml_rust/src/orm/snapshot.rs (hunk 9 above)
Comment (montanalow) [reply pgml-extension/pgml_rust/src/orm/snapshot.rs:282]: it's unsigned. Can't be < 0.

**Pull request merged**